crates/biome_service/src/file_handlers/astro.rs

```diff
@@ -0,0 +1,109 @@
+use crate::file_handlers::{
+    AnalyzerCapabilities, Capabilities, DebugCapabilities, ExtensionHandler, FormatterCapabilities,
+    Language, Mime, ParserCapabilities,
+};
+use crate::settings::SettingsHandle;
+use crate::WorkspaceError;
+use biome_formatter::Printed;
+use biome_fs::RomePath;
+use biome_js_formatter::format_node;
+use biome_js_parser::{parse_js_with_cache, JsParserOptions};
+use biome_js_syntax::{JsFileSource, JsLanguage};
+use biome_parser::AnyParse;
+use biome_rowan::{FileSource, NodeCache};
+use lazy_static::lazy_static;
+use regex::{Regex, RegexBuilder};
+use tracing::{debug, error, info};
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub(crate) struct AstroFileHandler;
+
+lazy_static! {
+    pub static ref ASTRO_FENCE: Regex = RegexBuilder::new(r#"^---\s*$"#)
+        .multi_line(true)
+        .build()
+        .unwrap();
```

Comment:
If astro file does not contain `---`, will it fail? (And it might be ok because it means there is no JavaScript?)
I was wondering if it is ok to use unwrap here.